client/src/app/videos/recommendations/recent-videos-recommendation.service.ts

```diff
@@ -14,26 +17,35 @@ export class RecentVideosRecommendationService implements RecommendationService
   readonly pageSize = 5
 
   constructor (
-    @Inject(VideoService) private videos: VideosProvider
+    private videos: VideoService,
+    private searchService: SearchService
   ) {
   }
 
-  getRecommendations (uuid: string): Observable<Video[]> {
-    return this.fetchPage(1)
+  getRecommendations (video: VideoDetails): Observable<Video[]> {
```

Comment:
As mentioned in [my PR](https://github.com/Jorropo/PeerTube/pull/6), I don't think using `VideoDetails` here is the right choice at this time because it's a very large object and we only use 2 of its properties.

I would recommend changing the signature to `(uuid: UUID, tags: string[])`.

Could also create a new class `RecommendationInfo` and pass that in, instead, if you think this function signature will swell greatly over time.